contracts/staking/StakingV1_1.sol

```diff
@@ -63,13 +143,31 @@ contract StakingV1_1 is StakingState, StakingConstants {
 
             IERC20(token).safeTransferFrom(msg.sender, address(this), stakeAmount);
 
+            //Deposit to sushi masterchef
+            if(token == LPToken){
+                uint256 sushiBalanceBefore = IERC20(SUSHI).balanceOf(address(this));
+                IMasterChefSushi(SUSHI_MASTERCHEF).deposit(
+                    BZRX_ETH_SUSHI_MASTERCHEF_PID,
+                    IERC20(LPToken).balanceOf(address(this))
+                );
+                uint256 sushiRewards = IERC20(SUSHI).balanceOf(address(this)) - sushiBalanceBefore;
```

Comment:
Since _addAltRewards is attributed to all LP stakers, this balanceBefore/balanceAfter calc might not be necessary. Could you instead just do: uint256 sushiRewards = IERC20(SUSHI).balanceOf(address(this));

Reply:
yes, true

Reply:
@eugenesh1 maybe what tom is implying `IERC20(SUSHI).balanceOf(address(this))` makes external call which we can save by keeping local balances ?

Reply:
but for that probably another call needs to be made `SUSHI_MASTERCHEF.getPendingReward` so yea ... balanceOf is good

Reply:
another note is ... we rely on sushi that balanceBefore <= current balance . need to check this in deep